backend/app/services/chat/chat_service.py

```
@@ -253,5 +253,51 @@ async def _handle_tool_calling_flow(
+        """
+        # Build messages
+        messages = message_builder.build_messages(
+            message=message,
+            system_prompt=system_prompt,
+            history=[],
+        )
+
+        # Get provider
+        provider = get_provider(model_config)
+        if not provider:
+            raise ValueError("Failed to create provider from model config")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)